dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/dataelement/DefaultCategoryService.java

```diff
@@ -132,6 +132,10 @@ public DefaultCategoryService( CategoryStore categoryStore, CategoryOptionStore
     @Transactional
     public long addCategory( Category dataElementCategory )
     {
+        if ( dataElementCategory.getShortName() == null )
```

Comment:
This I think we can skip since we set short names in the migration? There is also no guarantee that the name is less than 50 characters, and if we cut off the name at 50 characters, that it is still unique. Feels a bit hacky, better to remove I think.

Reply:
I added this mainly because of tests - many tests exists that indirectly use these methods and do not set a short name since it did not exist before. It would be lots of work to fix the test setup in the tests therefore I added this which helped with many of them. If you have any better idea I am all ears :)